internal/controller/controllers/agent_controller.go

```diff
@@ -1712,6 +1712,13 @@ func (r *AgentReconciler) setInfraEnvNameLabel(ctx context.Context, log logrus.F
 	return nil
 }
 
+func (r *AgentReconciler) setStateAnnotation(ctx context.Context, log logrus.FieldLogger, h *common.Host, agent *aiv1beta1.Agent) error {
```

Comment:
Do you really think this needs to be a separate function?

I'm not sure how much we're gaining by splitting this out.

Reply:
Well the Reconcile func is getting quite large:/ So for readability sake, wanted to keep it clean.
I'll try to remove it and let's see how it looks.